### PR TITLE
feat: extend fiveg_rfsim_interface

### PR DESCRIPTION
**Pull request by gatici**
# Description

This PR extends the fiveg_rfsim interface according to https://github.com/canonical/charm-relation-interfaces/pull/202 by putting RFSIM address, SST and SD information to the relation data bag.

- Remove sst and sd charm config options
- Do not use custom events (fiveg_rfsim_provider_available) instead use relation_changed event
- Modify and expand unit tests

Dependencies:

https://github.com/canonical/oai-ran-du-k8s-operator/pull/70 should be merged first.

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -126,18 +124,27 @@ def _configure(self, _) -> None:
...
         if service_restart_required := self._is_ue_config_up_to_date(ue_config):
             self._write_config_file(content=ue_config)
         self._configure_pebble(restart=service_restart_required)
 
+    @staticmethod
+    def get_sd_as_hex(value: Optional[int]) -> str:
+        """Convert the SD value to Hexadecimal if provided otherwise return default value."""
+        if value is None:
+            return "0x102030"
+        return hex(value)
```
[the remark this answers is not in the document]
Comment (gatici) [reply src/charm.py:146]: According to spec TE126, we decided to put the charm in `Waiting` status if Network Information (sst and sd(optional) is missing. Mainly charm waits for sst to be provided as sd is defined as an optional data https://github.com/canonical/sdcore-nms-k8s-operator/blob/main/lib/charms/sdcore_nms_k8s/v0/fiveg_core_gnb.py#L186. 

According to https://gitlab.eurecom.fr/oai/openairinterface5g/-/blob/develop/openair3/NAS/NR_UE/nr_nas_msg.c?ref_type=heads#L1294  and  https://gitlab.eurecom.fr/oai/openairinterface5g/-/blob/develop/openair3/NAS/NR_UE/nr_nas_msg.c?ref_type=heads#L1323, `sd` should exist.  

There could be two options here:
1. Similar to what this charm does previously, it uses a default sd value if this value is missing.
2.  We should change fiveg_core_gnb, five_f1 and fiveg_rfsim interfaces to mandate the sd availability.

[the remark this answers is not in the document]
Comment (patriciareinoso) [reply src/charm.py:146]: In the CU and DU , SD is not mandatory, we can create the configuration file with or without the SD

[the remark this answers is not in the document]
Comment (gatici) [reply src/charm.py]: Yes, this is done in latest commit.

File: tests/integration/test_integration.py [hunk 2]
```diff
@@ -161,21 +162,70 @@ async def _deploy_nrf(ops_test: OpsTest):
...
+                if not core_gnb_relations:
+                    logger.info("No %s relations found.", CORE_GNB_RELATION_NAME)
+                    return
+                for relation in core_gnb_relations:
+                    self._fiveg_core_gnb_provider.publish_gnb_config_information(
+                        relation_id=relation.id,
+                        tac=1,
+                        plmns=[PLMNConfig(mcc="001", mnc="01", sst=1, sd=1056816)],
+                    )
+        """
```
Comment (ghislainbourgeois) [inline tests/integration/test_integration.py:217]: I think this would be better in a separate file and reading it in for readability.

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: tests/integration/test_integration.py (hunk 2 above)
Comment (gatici) [reply tests/integration/test_integration.py]: This is moved to another file to increase readability.

File: src/charm.py [hunk 3]
```diff
@@ -170,8 +178,8 @@ def _generate_ue_config(self) -> str:
             key=self._charm_config.key,
             opc=self._charm_config.opc,
             dnn=self._charm_config.dnn,
-            sst=self._charm_config.sst,
-            sd=self._charm_config.sd,
+            sst=self.rfsim_requirer.sst,  # type: ignore[arg-type]
```
Comment (patriciareinoso) [inline src/charm.py:181]: is there a way to fix this static issue instead of ignoring it ?

**Review by patriciareinoso — COMMENTED** [review]
small comment added. Other that that. it looks good

File: src/charm.py (hunk 3 above)
Comment (gatici) [reply src/charm.py]: Static issue is fixed.

**Review by ghislainbourgeois — APPROVED** [review]

**Review by patriciareinoso — APPROVED** [review]

**Pull request merged**